graphdatascience/tests/unit/test_graph_construct.py

```diff
@@ -28,54 +26,22 @@ def test_graph_project_based_construct_without_arrow(runner: CollectingQueryRunn
     runner.add__mock_result("gds.debug.sysInfo", DataFrame([{"gdsEdition": "Unlicensed"}]))
     gds.graph.construct("hello", nodes, relationships, concurrency=2)
 
-    expected_node_query = "UNWIND $nodes as node RETURN node[0] as id, node[1] as labels, node[2] as propA"
-    expected_relationship_query = (
-        "UNWIND $relationships as relationship RETURN "
-        "relationship[0] as source, relationship[1] as target, "
-        "relationship[2] as type, relationship[3] as relPropA"
-    )
-    expected_proc_query = (
-        "CALL gds.graph.project.cypher("
-        "$graph_name, "
-        "$node_query, "
-        "$relationship_query, "
-        "{readConcurrency: $read_concurrency, parameters: { nodes: $nodes, relationships: $relationships }})"
+    expected_query = (
+        "UNWIND $data AS data WITH data, "
+        "CASE WHEN data[6] THEN data[5] ELSE null END AS sourceNodeLabels, "
+        "CASE WHEN data[3] THEN data[2] ELSE null END AS relationshipType, "
+        "CASE WHEN data[10] THEN data[9] ELSE null END AS targetNodeId, "
+        "\nCASE WHEN data[8] THEN data[7] ELSE null END AS sourceNodeProperties, "
+        "\nCASE WHEN data[1] THEN data[0] ELSE null END AS relationshipProperties "
+        "RETURN gds.graph.project($graph_name, data[4], targetNodeId, {sourceNodeLabels: sourceNodeLabels, targetNodeLabels: NULL, sourceNodeProperties: sourceNodeProperties, targetNodeProperties: NULL, relationshipType: relationshipType, relationshipProperties: relationshipProperties}, $configuration)"
```

Comment:
did you make the client use new Cypher projection in this PR? I cannot see any changes that correspond to that.

Reply:
I am thinking that your removal of the parameterised ServerVersion(2.1.0) makes `graph.construct` take a different path through the code.

I think we should follow this work with removals of code that is parameterised on ServerVersion older than 2.6. That would remove the old Cypher projection from `graph.construct` is my hypothesis. Otherwise I don't understand how things hang together.

Reply:
This is a unit-test and only worked based the old server-version.
Yes we can remove the old impl code as well, but IMO we should do this in 1.13.
We will raise awareness in 1.12 and act on 1.13? I assume many missed the docs about supported versions in the past